src/app/games/diaballik/DiaballikMoveGenerator.ts

```diff
@@ -75,71 +73,78 @@ class DiaballikMoveInConstruction {
             third = MGPOptional.of(this.subMoves[2]);
         }
         const move: DiaballikMove = new DiaballikMove(first, second, third);
-        Utils.assert(DiaballikRules.get().isLegal(move, state).isSuccess(),
+        Utils.assert(DiaballikRules.get().isLegal(move, this.stateBefore).isSuccess(),
                      'DiaballikMoveGenerator should only generate legal moves');
         return move;
     }
 }
 
+
 export class DiaballikMoveGenerator extends MoveGenerator<DiaballikMove, DiaballikState> {
 
     public getListMoves(node: DiaballikNode): DiaballikMove[] {
-        let movesInConstruction: DiaballikMoveInConstruction[] = [new DiaballikMoveInConstruction([], node.gameState)];
+        function finalize(m: DiaballikMoveInConstruction): DiaballikMove {
+            return m.finalize();
+        }
```

Comment:
double raison de staticiser finalize, même si Classe::methode devrais marcher comme en java !
Just for fun: dans les vrais fonctionnels, à quoi ça ressemblerais en statique vs methode ?

Reply:
Dans les vrais fonctionnels tu fais pas d'OO donc t'as pas de méthode, juste des fonctions !
Dans des fonctionnels-objets, bah tu ferais `foo.map(_.finalize)`

Reply:
_ servant de ref à l'object mappé ? pas super sexy.
Tiens, quelque pur fonctionnels vs fonctionnel-objet (en bien fait vs en mal fait) ?

Reply:
pur fonctionnel > fonctionnel-objet > objet, ceci étant des égalités strictes :p

Reply:
j'veux dire des exemples de langages types de chacun de ses trois groupes (enfin objet, j'ai :D haha)

Reply:
Pour des exemples de langages "purs" qui représentent bient leur modèle, on pourrait prendre Haskell pour fonctionnel, Scala pour fonctionnel-objet, Smalltalk pour objet !